py/tools/venv_shim/src/main.rs

```diff
@@ -148,30 +187,27 @@ fn main() -> miette::Result<()> {
         }
     };
 
-    let Some(interpreter_path) = python_executables.get(index) else {
+    let Some(actual_interpreter_path) = python_executables.get(index) else {
         return Err(miette!(
             "Unable to find another interpreter at index {}",
             index
         ));
     };
 
-    let exe_path = current_exe.to_string_lossy().into_owned();
     let exec_args = &args[1..];
 
     #[cfg(feature = "debug")]
     eprintln!(
         "[aspect] Attempting to execute: {:?} with argv[0] as {:?} and args as {:?}",
-        interpreter_path, exe_path, exec_args,
+        &actual_interpreter_path, &venv_interpreter_path, exec_args,
     );
 
-    let mut cmd = Command::new(&interpreter_path);
+    let mut cmd = Command::new(actual_interpreter_path);
     cmd.args(exec_args);
 
     // Lie about the value of argv0 to hoodwink the interpreter as to its
     // location on Linux-based platforms.
-    if cfg!(target_os = "linux") {
-        cmd.arg0(&exe_path);
-    }
+    cmd.arg0(&venv_interpreter_path);
```

Comment:
hmm, should this be cfg! not target_os  = "macos"?

Reply:
Testing shows that it doesn't have to be.